File: app/src/main/java/spa/lyh/cn/permission/adapter/PhotoAdapter.java
package spa.lyh.cn.permission.adapter;

import android.util.Log;
import android.widget.BaseAdapter;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

import java.util.List;

import spa.lyh.cn.lib_image.app.ImageLoadUtil;
import spa.lyh.cn.permission.R;
import spa.lyh.cn.permission.bean.PictureBean;

public class PhotoAdapter extends BaseQuickAdapter<PictureBean, BaseViewHolder> {
    public PhotoAdapter(@Nullable List<PictureBean> data) {
        super(R.layout.item_photo, data);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder baseViewHolder, PictureBean pictureBean) {
        ImageView img = baseViewHolder.getView(R.id.photo);
        ImageLoadUtil.displayImage(getContext(),pictureBean.getFirstImageUrl(),img);
    }
}

File: app/src/main/java/spa/lyh/cn/permission/adapter/PhotoAdapter.kt
package spa.lyh.cn.permission.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.chad.library.adapter4.BaseQuickAdapter
import spa.lyh.cn.lib_image.app.ImageLoadUtil
import spa.lyh.cn.permission.bean.PictureBean
import spa.lyh.cn.permission.databinding.ItemPhotoBinding

class PhotoAdapter(data: List<PictureBean>) :
    BaseQuickAdapter<PictureBean, PhotoAdapter.VH>(data) {

    class VH(val b:ItemPhotoBinding):ViewHolder(b.root)

    override fun onBindViewHolder(holder: VH, position: Int, item: PictureBean?) {
        ImageLoadUtil.displayImage(context, item!!.firstImageUrl, holder.b.photo)
    }

    override fun onCreateViewHolder(context: Context, parent: ViewGroup, viewType: Int): VH {
        return VH(ItemPhotoBinding.inflate(LayoutInflater.from(context),parent,false))
    }
}
